targets/TARGET_NORDIC/TARGET_MCU_NRF51822/sleep.c

```diff
@@ -18,16 +18,16 @@
 #include "mbed_interface.h"
 #include "toolchain.h"
 
-MBED_WEAK void sleep(void)
+MBED_WEAK void hal_sleep(void)
 {
     // ensure debug is disconnected if semihost is enabled....
     NRF_POWER->TASKS_LOWPWR = 1;
     // wait for interrupt
     __WFE();
 }
 
-MBED_WEAK void deepsleep(void)
+MBED_WEAK void hal_deepsleep(void)
```

Comment:
We might want to remove weak linkage, as it's not applied to any other implementations.